psyneulink/library/components/mechanisms/processing/integrator/ddm.py

```diff
@@ -1195,3 +1195,26 @@ def is_finished(self, context=None):
             )
             return True
         return False
+    
+    def _gen_llvm_is_finished_cond(self, ctx, builder, params, state, current):
+        # setup pointers to internal function
+        func_state_ptr = pnlvm.helpers.get_state_ptr(builder, self, state, 'function')
+        func_param_ptr = pnlvm.helpers.get_state_ptr(builder, self, params, 'function')
+        
+        # load prev value
+        try:
+            prev_val_ptr = pnlvm.helpers.get_state_ptr(builder, self.function, func_state_ptr, "previous_value")
+        except ValueError:
+            return pnlvm.ir.IntType(1)(1)
+        prev_val_ptr = builder.gep(prev_val_ptr, [ctx.int32_ty(0), ctx.int32_ty(0), ctx.int32_ty(0)])
+        prev_val = builder.load(prev_val_ptr)
+
+        # obtain threshold value
+        threshold_ptr = pnlvm.helpers.get_param_ptr(builder, self.function, func_param_ptr,
+                                          "threshold")
+        threshold_ptr = builder.gep(threshold_ptr, [ctx.int32_ty(0), ctx.int32_ty(0)])
+        threshold = builder.load(threshold_ptr)
+        is_prev_greater = builder.fcmp_ordered('>=', prev_val, threshold)
```

Comment:
this is not 'greater', but 'greater_or_equal'. The python version also uses `abs(prev_value)`.

Reply:
is there a builtin version of abs?

Reply:
you can use llvm intrinsic: `ctx.get_builtin("fabs", [ctx.float_ty])`